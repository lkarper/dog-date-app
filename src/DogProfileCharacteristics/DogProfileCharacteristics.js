import React from 'react';
import PropTypes from 'prop-types';

const DogProfileCharacteristics = (props) => {

    const { dog_profile } = props;

    const {
        energy,
        temperment,
        obedience,
        dislikes_puppies,
        dislikes_men,
        dislikes_women,
        dislikes_children,
        recently_adopted,
        prefers_people,
        leash_aggression,
        elderly_dog,
        little_time_with_other_dogs,
        much_experience_with_other_dogs,
        aggressive,
    } = dog_profile;

    if (!energy || ! temperment || !obedience) {
        return (
            <ul>
                <li>Error: Could not load characteristics.  Check your connection and the URL and try again.</li>
            </ul>
        );
    }

    return (
        <ul>
            <li>{energy === 'Not very...' ? 'Not very energetic...' : energy}</li>
            <li>{temperment}</li>
            <li>{obedience}</li>
            {dislikes_puppies ? <li>Dislikes puppies</li> : ''}
            {dislikes_men ? <li>Dislikes men</li> : ''}
            {dislikes_women ? <li>Dislikes women</li> : ''}
            {dislikes_children ? <li>Dislikes children</li> : ''}
            {recently_adopted ? <li>Recently adopted</li> : ''}
            {prefers_people ? <li>Prefers people to dogs</li> : ''}
            {leash_aggression ? <li>Has leash aggression</li> : ''}
            {elderly_dog ? <li>Elderly dog</li> : ''}
            {little_time_with_other_dogs ? <li>Has spent little time with other dogs</li> : ''}
            {much_experience_with_other_dogs ? <li>Has much experience with other dogs</li> : ''}
            {aggressive ? <li>Aggressive dog</li> : ''}
        </ul>
    );

}

DogProfileCharacteristics.defaultProps = {
    dog_profile: {
        energy: '',
        temperment: '',
        obedience: '',
        dislikes_puppies: false,
        dislikes_men: false,
        dislikes_women: false,
        dislikes_children: false,
        recently_adopted: false,
        prefers_people: false,
        leash_aggression: false,
        elderly_dog: false,
        little_time_with_other_dogs: false,
        much_experience_with_other_dogs: false,
        aggressive: false,
    },
};

DogProfileCharacteristics.propTypes = {
    dog_profile: PropTypes.shape({
        energy: PropTypes.string,
        temperment: PropTypes.string,
        obedience: PropTypes.string,
        dislikes_puppies: PropTypes.bool,
        dislikes_men: PropTypes.bool,
        dislikes_women: PropTypes.bool,
        dislikes_children: PropTypes.bool,
        recently_adopted: PropTypes.bool,
        prefers_people: PropTypes.bool,
        leash_aggression: PropTypes.bool,
        elderly_dog: PropTypes.bool,
        little_time_with_other_dogs: PropTypes.bool,
        much_experience_with_other_dogs: PropTypes.bool,
        aggressive: PropTypes.bool,
    }).isRequired,
}

export default DogProfileCharacteristics;
