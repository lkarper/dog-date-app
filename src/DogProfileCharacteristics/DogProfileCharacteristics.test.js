import React from 'react';
import ReactDOM from 'react-dom';
import { mount } from 'enzyme';
import toJSON from 'enzyme-to-json';
import DogProfileCharacteristics from './DogProfileCharacteristics';

describe('DogProfileCharacteristics component', () => {
    it('renders the UI without crashing', () => {
        const div = document.createElement('div');
        ReactDOM.render(<DogProfileCharacteristics />, div);
        ReactDOM.unmountComponentAtNode(div);
    });

    it('renders the UI as expected', () => {
        const wrapper = mount(<DogProfileCharacteristics />);
        expect(toJSON(wrapper)).toMatchSnapshot();
        wrapper.unmount();
    });
});
