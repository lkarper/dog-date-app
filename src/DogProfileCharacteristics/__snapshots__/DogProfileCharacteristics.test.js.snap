// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DogProfileCharacteristics component renders the UI as expected 1`] = `
<DogProfileCharacteristics
  dog_profile={
    Object {
      "aggressive": false,
      "dislikes_children": false,
      "dislikes_men": false,
      "dislikes_puppies": false,
      "dislikes_women": false,
      "elderly_dog": false,
      "energy": "",
      "leash_aggression": false,
      "little_time_with_other_dogs": false,
      "much_experience_with_other_dogs": false,
      "obedience": "",
      "prefers_people": false,
      "recently_adopted": false,
      "temperment": "",
    }
  }
>
  <ul>
    <li>
      Error: Could not load characteristics.  Check your connection and the URL and try again.
    </li>
  </ul>
</DogProfileCharacteristics>
`;
